dist/index.js

```diff
@@ -23769,31 +23769,33 @@ async function run() {
       continue;
     }
     core.debug(
-      `Linked issues for PR #${pr.prNumber}: ${pr.issues.map((issue) => issue.number).join(",")}`
+      `Linked issues for PR #${pr.prNumber}: ${pr.issues.map((issue) => `${owner}/${issue.repository.name}#${issue.number}`).join(",")}`
     );
     for (const issue of pr.issues) {
+      const issueRepo = issue.repository.name;
       if (await hasExistingComment(octokit, {
-        repo,
+        repo: issueRepo,
         owner,
         issueNumber: issue.number
       })) {
         core.debug(`Comment already exists for issue #${issue.number}`);
         continue;
       }
+      const prLink = repo === issueRepo ? `#${pr.prNumber}` : `[#${pr.prNumber}](https://github.com/${owner}/${repo}/pull/${pr.prNumber})`;
       const body = `${RELEASE_COMMENT_HEADING}
 
-This issue was referenced by PR #${pr.prNumber}, which was included in the [${version2} release](https://github.com/${owner}/${repo}/releases/tag/${version2}).`;
+This issue was referenced by PR ${prLink}, which was included in the [${version2} release](https://github.com/${owner}/${repo}/releases/tag/${version2}).`;
       core.debug(`Creating comment for issue #${issue.number}`);
       try {
         await octokit.rest.issues.createComment({
-          repo,
+          repo: issueRepo,
           owner,
```

Comment:
owner is also important I think!

Reply:
added a comment here, we enforce the same owner here as the release!